cmd/juju/controller/addmodel.go

```diff
@@ -36,15 +37,17 @@ func NewAddModelCommand() cmd.Command {
 		newCloudAPI: func(caller base.APICallCloser) CloudAPI {
 			return cloudapi.NewClient(caller)
 		},
+		providerRegistry: environs.GlobalProviderRegistry(),
 	})
 }
 
 // addModelCommand calls the API to add a new model.
 type addModelCommand struct {
 	modelcmd.ControllerCommandBase
-	apiRoot        api.Connection
-	newAddModelAPI func(base.APICallCloser) AddModelAPI
-	newCloudAPI    func(base.APICallCloser) CloudAPI
+	apiRoot          api.Connection
+	newAddModelAPI   func(base.APICallCloser) AddModelAPI
+	newCloudAPI      func(base.APICallCloser) CloudAPI
+	providerRegistry environs.ProviderRegistry
```

Comment:
It's a shame we need to add provider dependency to another CLI command, but necessary it seems. We did have a goal to say release a CLI binary minus bootstrap which was much slimmer is size.